### PR TITLE
ENH: add `ExtensionArray._explode` method; adjust pyarrow extension for use of new interface

### LINKED ISSUE #54833: ENH: make `_explode` a method of the `ExtensionArray` interface
### Feature Type

- [X] Adding new functionality to pandas

- [X] Changing existing functionality in pandas

- [ ] Removing existing functionality in pandas

### Problem Description

In awkward-pandas we've had a user expect `explode` to work for them; see https://github.com/intake/awkward-pandas/issues/38

I see in https://github.com/pandas-dev/pandas/pull/53602/ `Series.explode` support for extension types has been added specifically to PyArrow list types. We'd like to be able to support `explode` with our `awkward` extension type. Therefore this feature request is to make an `ExtensionArray._explode` method, with a default that provides the same behavior already provided by `Series.explode` for non object dtypes (that is, a copy).

### Feature Description

Issue will come with accompanying PR.

### Alternative Solutions

None to my knowledge

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by douglasdavis**
- [x] closes #54833 
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [x] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/arrays/arrow/array.py [hunk 1]
```diff
@@ -1609,6 +1609,10 @@ def _explode(self):
         """
         See Series.explode.__doc__.
         """
+        # child class explode method supports only list types; return
+        # default implementation for non list types.
+        if self.dtype.type != list:
```
Comment (mroeschke) [inline pandas/core/arrays/arrow/array.py:1614]:
```suggestion
        if not pa.types.is_list(self.dtype.pyarrow_dtype):
```

File: doc/source/whatsnew/v2.1.0.rst [hunk 2]
```diff
@@ -246,6 +246,7 @@ Other enhancements
 - :meth:`.SeriesGroupBy.transform` and :meth:`.DataFrameGroupBy.transform` now support passing in a string as the function for ``engine="numba"`` (:issue:`53579`)
 - :meth:`DataFrame.stack` gained the ``sort`` keyword to dictate whether the resulting :class:`MultiIndex` levels are sorted (:issue:`15105`)
 - :meth:`DataFrame.unstack` gained the ``sort`` keyword to dictate whether the resulting :class:`MultiIndex` levels are sorted (:issue:`15105`)
+- :meth:`ExtensionArray._explode` interface method added to allow extension type implementations of the ``explode`` method (:issue:`54833`)
```
[the remark this answers is not in the document]
Comment (douglasdavis) [reply doc/source/whatsnew/v2.1.0.rst:249]: yep!

**Comment by mroeschke** [comment]
```bash
Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:1928:GL01:pandas.api.extensions.ExtensionArray._explode:Docstring text (summary) should start in the line immediately after the opening quotes (not in the same line, or leaving a blank line in between)
Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:1928:GL02:pandas.api.extensions.ExtensionArray._explode:Closing quotes should be placed in the line after the last text in the docstring (do not close the quotes in the same line as the text, or leave a blank line between the last text and the quotes)
Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:1928:GL03:pandas.api.extensions.ExtensionArray._explode:Double line break found; please use only one blank line to separate sections or paragraphs, and do not leave blank lines at the end of docstrings
Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:1928:SA03:pandas.api.extensions.ExtensionArray._explode:Description should be capitalized for See Also "Series.explode" reference
Error: /home/runner/work/pandas/pandas/pandas/core/arrays/base.py:1928:EX01:pandas.api.extensions.ExtensionArray._explode:No examples section found
```

**Review by mroeschke — APPROVED** [review]
LGTM cc @jbrockmendel if you have thoughts

**Comment by douglasdavis** [comment]
yep, fixed! (CI failures look unrelated; spelling in a file untouched but this PR and an installation issue associated with BLAS)

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @douglasdavis

**Pull request merged**